lib/mysql.js

```diff
@@ -60,7 +66,39 @@ function defineMySQLTypes(dataSource) {
  */
 function MySQL(settings) {
   SqlConnector.call(this, 'mysql', settings);
+}
+
+require('util').inherits(MySQL, SqlConnector);
 
+MySQL.prototype.connect = function(callback) {
+  var self = this;
+  var options = generateOptions(this.settings);
+  var s = self.settings || {};
+
+  if (this.client) {
+    if (callback) {
+      process.nextTick(function() {
+        callback(null, self.client);
+      });
+    }
+  } else {
+    this.client = mysql.createPool(options);
+    this.client.getConnection(function(err, connection) {
```

Comment:
@raymondfeng Yeah, the thing is if I use the original code, no `error` event would be triggered:
```js
this.client = mysql.createPool(options);

  this.client.on('error', function(err) {
    dataSource.emit('error', err);
    dataSource.connected = false;
    dataSource.connecting = false;
  });
```

So I have to call `getConnection` to get the connection error message.